### ISSUE TITLE
platform_version error when compiling for iOS using SDK 50.2.3.6

### ISSUE THREAD

**Issue by psvdwal** [airsdk/Adobe-Runtime-Support#2885]
I tried to compile an existing actionscript project for iOS, using AIR SDK version 50.2.3.6 and got the error message: 
"Error creating files. Id: Unknown option: -platform_version. Compilation failed while executing: Id64."

The same actionscript project compiles successfully when using AIR SDK version 50.2.2.2 for iOS. 

I use no external ane's. Just plain actionscript code. 

Any suggestion what's wrong?
Kind regards,
Piet van der Wal
![Screenshot_AIRSDK_error_platform_version_iOS](https://github.com/airsdk/Adobe-Runtime-Support/assets/7584108/7e35cad4-b6a0-4636-9b70-d72339fab072)

**Comment by ajwfrost** [comment]
Hi

The linker (ld64.exe) needs to be updated for this, we've got a version that you can download from the below link:
https://transfer.harman.com/link/Brh2YO2NAELPL99TZbgimo
If you unzip that and copy the files over the ones in the AIR SDK "lib/aot/bin/ld64" folder it should then work..

thanks

**Comment by psvdwal** [comment]
Hi Andrew,

 

That solved the problem. Thanks!

 

Best,

Piet

**Comment by xabip** [comment]
Hi Andrew, 

I've updated the ld64 using your package (working successfully, but I still need to select the specific -platformsdk for iOS to make it work). 
However, checking the debug logs, when the lb64 is invoked, it's targeting an older 16.4 platform_version like this "-platform_version ios 14.0 16.4".  Can this be updated? Can any parameter in the application descriptor or ADT be passed? 

Best

**Comment by ajwfrost** [comment]
> but I still need to select the specific -platformsdk for iOS to make it work

Ah, I think on iOS the ADT tool is still using the default system `ld` command which would be why you get problems with Xcode 15 unless using the direct iPhoneOS SDK... that will change soon!

The commands passed to the linker are based on the ADT settings, and the build set-up that we used for the release (see section 2.2.3 in the release notes). So for that version:
- minimum OS level is 11.0 by default, but you (or ANEs) can adjust that in InfoAdditions or linker commands
- SDK used is whatever we built the stub files using, so 16.4 for this build

I think the better way would for us to change the SDK version if it's not using our stubs, so that it just picked up the details from the `-platformsdk` path. If that's actually possible, we would have to check (although you may have seen a warning if you're using an SDK that doesn't end in the version code, so we maybe could use that same mechanism..)

There may be some odd effects though if we allowed completely free reign over whatever values were going to be passed in there..

**Comment by Mintonist** [comment]
@ajwfrost 
```
The linker (ld64.exe) needs to be updated for this, we've got a version that you can download from the below link:
https://transfer.harman.com/link/Brh2YO2NAELPL99TZbgimo
```

Does it mean that I can continue build .ipa releases on my Windows PC?

**Comment by ajwfrost** [comment]
Yes, with this new linker, you can continue with IPA creation on Windows even with libraries that were created using Xcode 14. With a lot of thanks to the clever people at LLVM!
